bliss/unconstrained_dists.py

```diff
@@ -32,15 +33,49 @@ def get_dist(self, params):
         return Normal(mean, sd)
 
 
-class UnconstrainedDiagonalBivariateNormal:
+class TruncatedDiagonalMVN(Distribution):
+    """A truncated diagonal multivariate normal distribution."""
+
+    def __init__(self, mu, sigma):
+        super().__init__()
+
+        multiple_normals = Normal(mu, sigma)
+        prob_in_unit_box_hw = multiple_normals.cdf(torch.ones_like(mu))
+        prob_in_unit_box_hw -= multiple_normals.cdf(torch.zeros_like(mu))
+        self.log_prob_in_unit_box = prob_in_unit_box_hw.log().sum(dim=-1)
+
+        self.base_dist = Independent(multiple_normals, 1)
+
+    def sample(self, **args):
+        # some ideas for how to sample it here, if we need to:
+        # https://cran.r-project.org/web/packages/truncnorm/
+        raise NotImplementedError("sampling a truncated normal isn't straightforward")
```

Comment:
A question related to #806 - would we ever realistically need to sample from the variational distribution (since we've been talking about it)? Because this would prohibit that from working. If we just want to show how to sample without actually needing to implement it then this should be fine.